dcos_test_utils/gce.py

```diff
@@ -0,0 +1,174 @@
+import os
+import json
+import logging
+from googleapiclient.errors import HttpError
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+from retrying import retry
+
+log = logging.getLogger(__name__)
+
+
+def catch_http_exceptions(f):
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if 'HttpError 404' in str(e):
+                raise Exception("The resource you are trying to access doesn't exist")
+            elif 'HttpError 409' in str(e):
+                raise Exception('''The resource you are trying to access is either being created (operation conflict) or
+                    already exists''')
+
+    return handle_exception
+
+
+class GceWrapper:
+    def __init__(self, credentials_path=None):
+        if credentials_path is None:
+            if 'GOOGLE_APPLICATION_CREDENTIALS' not in os.environ:
+                raise Exception('''You must either set the GOOGLE_APPLICATION_CREDENTIALS environment
+                    variable or supply a credentials path to the GceWrapper constructor.''')    
+        elif not os.path.isfile(credentials_path):
+            raise Exception(credentials_path + ' is not a valid google credentials path.')
+        else:
+            os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = credentials_path
+
+        credentials = GoogleCredentials.get_application_default()
```

Comment:
@mellenburg GoogleCredentials.get_application_default() creates a credentials object from the GOOGLE_APPLICATION_CREDENTIALS variable which was being set to credentials_path (but not anymore)